can/notifier.py

```diff
@@ -125,15 +125,15 @@ def _rx_thread(self, bus: BusABC):
             elif not self._on_error(exc):
                 raise
 
-    def _on_message_available(self, bus: BusABC):
+    def _on_message_available(self, bus: BusABC) -> None:
         msg = bus.recv(0)
         if msg is not None:
             self._on_message_received(msg)
 
-    def _on_message_received(self, msg: Message):
+    def _on_message_received(self, msg: Message) -> None:
         for callback in self.listeners:
-            res = callback(msg)
-            if self._loop is not None and asyncio.iscoroutine(res):
+            res = cast(Optional[Awaitable[Any]], callback(msg))
+            if res is not None and self._loop is not None and asyncio.iscoroutine(res):
```

Comment:
Is this always the case? The result of calling the `callback` (for listeners that just calls `on_message_received`) is an Awaitable?

Reply:
Nope, good catch. As the next line states, it can also be `None`.

```suggestion
            res = cast(Union[None, Optional[Awaitable[Any]]], callback(msg))
            if res is not None and self._loop is not None and asyncio.iscoroutine(res):
```